### PR TITLE
`Exam mode`: Prevent problem statement from closing when viewing changes

### PR DESCRIPTION
**Pull request by pzdr7**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).

### Motivation and Context

During the second testing session for 7.0.0, we noticed that clicking on the "Show Difference" / "Only show new Version" buttons closed the problem statement for some exercise types.

This was because the problem statement (which collapses when clicked) registered a click when the button was clicked.

### Description

- Stops the click from propagating to the problem statement, thus preventing it from collapsing
- Verifies in the tests that this `stopPropagation` occurs

### Steps for Testing

Prerequisites:
- 1 Exam (this change should affect all exercise types except for programming, which doesn't have this issue)
- 1 Student
- 1 Instructor

1. Open the exam as a student, viewing an exercise.
2. In a different browser: As the instructor, edit the problem statement of that exercise.
3. When the option becomes available, click on "Show Difference" / "Only show new Version"
4. Verify that the problem statement does not collapse when you click on the button.
5. Verify that the problem statement still collapses when you click elsewhere on the problem statement.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/main/webapp/app/exam/participate/exercises/exam-exercise-update-highlighter/exam-exercise-update-highlighter.component.ts [hunk 1]
```diff
@@ -37,7 +37,8 @@ export class ExamExerciseUpdateHighlighterComponent implements OnInit, OnDestroy
      * Switches the view between the new(updated) problem statement without the difference
      * with the view showing the difference between the new and old problem statement and vice versa.
      */
-    toggleHighlightedProblemStatement(): void {
+    toggleHighlightedProblemStatement(event: MouseEvent): void {
+        event.stopPropagation();
```
Comment (julian-christl) [inline src/main/webapp/app/exam/participate/exercises/exam-exercise-update-highlighter/exam-exercise-update-highlighter.component.ts:41]:
```suggestion
        // prevents the jhi-resizeable-container from collapsing the right panel on a button click
        event.stopPropagation();
```

**Review by julian-christl — APPROVED** [review]
code lgtm

**Review by coolchock — APPROVED** [review]
tested during testing session, works as described. Code LGTM

**Review by undernagruzez — APPROVED** [review]
tested during a testing session and works as expected, code also lgtm

**Review by muradium — APPROVED** [review]
Tested during testing session, works as expected

**Review by rstief — APPROVED** [review]
tested in testing session, works as expected.

**Pull request merged**